security/pkg/server/ca/server.go

```diff
@@ -214,8 +229,9 @@ func New(ca ca.CertificateAuthority, ttl time.Duration, forCA bool, hostlist []s
 	}
 
 	version.Info.RecordComponentBuildTag("citadel")
+	rootCertExpiryTime.Set(rootCertExpirationSeconds(ca))
```

Comment:
does this mean ```rootCertExpirationSeconds``` is only called in ```New``` (when citadel start) ?
(I was looking at places pilot do this, ex [monServices.Set(float64(len(push.Services(nil))))](https://github.com/istio/istio/blob/49f3d52d3b8c6258561018118f316c718403d4e2/pilot/pkg/proxy/envoy/v2/ads.go#L805) in called in each push. )

Reply:
Yes. I think that matches the original FR description, a value will be set only once.

https://github.com/istio/istio/issues/14526#issue-451502343

Reply:
what if citadel run sometime then root get expired, we won't get updated metric ?

Reply:
correct, the metric is a constant unless root is updated && citadel is restarted. customer will configure alert as `citadel_root_cert_expiry_time_seconds - time() < 0`.

I probably should rename the variable name to avoid confusion?

Reply:
probably just print out timestamp when root cert is expired ? (without calculating duration)

Reply:
Yes, we already have a error log line, https://github.com/istio/istio/pull/14569/files/681731be33b69e09ed12d05da05e294417778bf4#diff-5b0193090dbec715ed170297027bb7bcR114